### PR TITLE
[4.0][a11y] 'Cancel' button changes from an anchor tag to a button tag (profile edit)

### PR DESCRIPTION
**Pull request by tarot-ray**
For A11Y reasons, this should be a button tag, not a styled anchor tag.

Pull Request for Issue # .

### Summary of Changes

'Cancel' button is changed to a button tag from an anchor tag, to improved A11Y

### Testing Instructions

Check that the 'Cancel' button is seen correctly by a screenreader.

### Expected result

Improved A11Y.

### CODE REVIEW

**Comment by SharkyKZ** [comment]
This doesn't work. Button with `type="button"` needs some JS code to do anything. `href` is not a valid attribute for element `button`. If this really needs to be a button it should submit the form with cancel task.

File: components/com_users/tmpl/profile/edit.php [hunk 1]
```diff
@@ -129,7 +129,7 @@
 						<?php echo Text::_('JSUBMIT'); ?>
 					</span>
 				</button>
-				<a class="btn btn-danger" href="<?php echo Route::_('index.php?option=com_users&view=profile'); ?>" title="<?php echo Text::_('JCANCEL'); ?>"><?php echo Text::_('JCANCEL'); ?></a>
```
Comment (wilsonge) [inline components/com_users/tmpl/profile/edit.php:132]: We still need the classes here

**Review by wilsonge — CHANGES_REQUESTED** [review]
As per the comments

**Comment by wilsonge** [comment]
I have tested this item :red_circle: unsuccessfully on 9ce88b685e30ebab5e4fec15da83dc79fd8698ad<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24007">issues.joomla.org/tracker/joomla-cms/24007</a>.</sub>

File: components/com_users/tmpl/profile/edit.php (hunk 1 above)
Comment (tarot-ray) [reply components/com_users/tmpl/profile/edit.php:132]: Done.

**Comment by wilsonge** [comment]
You still need to address the comments from @SharkyKZ

**Review by wilsonge — APPROVED** [review]

**Comment by wilsonge** [comment]
I have tested this item :white_check_mark: successfully on 0b3c29fb929d0a6c184b01132ebcf2e4afdd5ade<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24007">issues.joomla.org/tracker/joomla-cms/24007</a>.</sub>

**Comment by wilsonge** [comment]
@tarot-ray almost there - in the latest commit you've accidentally removed the classes again :)

**Comment by zwiastunsw** [comment]
@tarot-ray: Add: `class="btn btn-danger"`

**Comment by tarot-ray** [comment]
So, further changes have been made to the PR to make the line absolutely correct and perfectly A11Y compatible. Please, review.

**Comment by zwiastunsw** [comment]
I have tested this item :white_check_mark: successfully on 74c891e5d33e05df813e26a45a894e697d8087fa<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24007">issues.joomla.org/tracker/joomla-cms/24007</a>.</sub>

**Comment by wilsonge** [comment]
LGTM!

**Comment by wilsonge** [comment]
I have tested this item :white_check_mark: successfully on 30dd62a410e3e7a070aa713aab417b592152c119<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24007">issues.joomla.org/tracker/joomla-cms/24007</a>.</sub>

**Comment by wilsonge** [comment]
Just whitespace changes between the last two tests so RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24007">issues.joomla.org/tracker/joomla-cms/24007</a>.</sub>

**Comment by SharkyKZ** [comment]
What exactly is the point of changing anchor to a button that does exactly the same but with JS dependency?

**Comment by wilsonge** [comment]
@zwiastunsw Actually in this specific case thinking about it - I think a link was the correct choice because this is a redirect to another page. From http://web-accessibility.carnegiemuseums.org/content/buttons/ 

```
Navigation: If you want a user to navigate to a new page or to a different target on the same page, use an anchor element <a>.
```

Basically buttons should be used if you're doing stuff on the same page and links for moving to another page

**Comment by zwiastunsw** [comment]
@SharkyKZ , @wilsonge :
It's a bit more complicated. In this case, Cancel will actually redirect to another page. But the essence of this control is to cancel the action and not to move user to another page.
In this and other similar cases, Cancel cancels actions you have performed. 
The command also speaks: **Cancel**, not **Go to**. 
This is not just `a` tag problem. If you use `a` tag, make it look like a link and not like a button. Because it's confusing for keyboard users. I run the link with the Enter key, the button I run with the Enter or Space keys. If you activating link with the spacebar causes the page to scroll and not running the link.

**Comment by SharkyKZ** [comment]
See #24088 for potential alternative please.

**Comment by ghost** [comment]
@wilsonge can this PR be closed?

**Comment by wilsonge** [comment]
I've merged https://github.com/joomla/joomla-cms/pull/24088 so closing this

**Pull request closed without merging**